pkg/timelock/timelock.go

```diff
@@ -166,22 +181,30 @@ func (tw *Worker) Listen() error {
 	shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
 
-	<-isclosed.All(shutdownCtx, schedulingDone, historyDone, newDone, processingDone)
+	<-isclosed.All(shutdownCtx, schedulingDone, historyDone, newDone, processingDone) //nolint:contextcheck
 
 	return nil
 }
 
 // setupFilterQuery returns an ethereum.FilterQuery initialized to watch the Timelock contract.
-func (tw *Worker) setupFilterQuery() ethereum.FilterQuery {
+func (tw *Worker) setupFilterQuery(fromBlock *big.Int) ethereum.FilterQuery {
 	return ethereum.FilterQuery{
 		Addresses: tw.address,
-		FromBlock: tw.fromBlock,
+		FromBlock: fromBlock,
+	}
+}
+
+func (tw *Worker) retrieveNewLogs(ctx context.Context) (<-chan struct{}, <-chan types.Log, error) {
+	if tw.ethClient.Client().SupportsSubscriptions() {
```

Comment:
this is where decide whether to use subscriptions (ws) or polling (http)

Reply:
Can we add this info as a comment on the function please?

Reply:
done -- pushed in the [latest update](3a065155b6aadba9b510e98831e14cd68d5ff1f9)